internal/monitors/resourcesync/resourcesync.go

```diff
@@ -0,0 +1,289 @@
+			if err != nil {
+				log.Errorf("Failed to update resourcesync status for %s: %v", rs.Name, err)
+			}
+			continue
+
+		}
+		addResourceParseCondition(rs, nil)
+		err = r.store.GetResourceSyncStore().UpdateStatusIgnoreOrg(rs)
+		if err != nil {
+			log.Errorf("Failed to update resourcesync status for %s: %v", rs.Name, err)
+			continue
+		}
+		syncLinks = append(syncLinks, syncLink{
+			resources:    resources,
+			hash:         hash,
+			resourceSync: *rs,
+		})
+	}
+
+	// We now have a list of generic resources and resourcesync - we can attempt to apply and set conditions
+	for i := range syncLinks { // Check if new commit hash
+		link := syncLinks[i]
+		var err error
+		for _, genResource := range link.resources {
+			err = r.applyResource(genResource, &link.resourceSync)
+			if err != nil {
+				log.Errorf("resourcesync/%s: Failed to apply resource. Error: %s", link.resourceSync.Name, err.Error())
+				break
+			}
+		}
+		if err != nil {
+			break
+		}
+		link.resourceSync.Status.Data.LastSyncedCommitHash = &link.hash
+		err = r.store.GetResourceSyncStore().UpdateStatusIgnoreOrg(&link.resourceSync)
+		if err != nil {
+			log.Errorf("Failed to update resourcesync status for %s: %v", link.resourceSync.Name, err)
+		}
+	}
+}
+func (r *ResourceSync) cloneRepo(repo *model.Repository, revision *string) (billy.Filesystem, string, error) {
+	storage := gitmemory.NewStorage()
+	mfs := memfs.New()
+	ops := &git.CloneOptions{
+		URL:   *repo.Spec.Data.Repo,
+		Depth: 1,
+	}
+	if repo.Spec.Data.Username != nil && repo.Spec.Data.Password != nil {
+		ops.Auth = &githttp.BasicAuth{
+			Username: *repo.Spec.Data.Username,
+			Password: *repo.Spec.Data.Password,
+		}
+	}
+	if revision != nil {
+		ops.ReferenceName = gitplumbing.ReferenceName(*revision)
+	}
+	gitRepo, err := git.Clone(storage, mfs, ops)
+	if err != nil {
+		return nil, "", err
+	}
+	head, err := gitRepo.Head()
+	if err != nil {
+		return nil, "", err
+	}
+	hash := head.Hash().String()
+	return mfs, hash, nil
+}
+
+func (r *ResourceSync) extractResourcesFromDir(orgId string, mfs billy.Filesystem, path string) ([]genericResource, error) {
+	genericResources := []genericResource{}
+	files, err := mfs.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range files {
+		if !file.IsDir() && isValidFile(file.Name()) { // Not going recursivly into subfolders
+			resources, err := r.extractResourcesFromFile(orgId, mfs, mfs.Join(path, file.Name()))
+			if err != nil {
+				return nil, err
+			}
+			genericResources = append(genericResources, resources...)
+		}
+	}
+	return genericResources, nil
+}
+
+func (r *ResourceSync) extractResourcesFromFile(orgId string, mfs billy.Filesystem, path string) ([]genericResource, error) {
+	genericResources := []genericResource{}
+
+	file, err := mfs.Open(path)
+	if err != nil {
+		// Failed to open file....
+		return nil, err
+	}
+	defer file.Close()
+	decoder := yamlutil.NewYAMLOrJSONDecoder(file, 100)
```

Comment:
Just curious what was the reason for the 100 size buffer. Probably be good to add a test and or docs around this decoding.

Reply:
no reason, looked at the cli (cmd), did the same). Any suggestions? Its still WIP, so please do suggest. :)

Reply:
It would be good to understand the size of files we expect to process. Broken down into 100 byte increments is this efficient? Hard to know without more details.

Reply:
Agree, we can discuss this in our next design meeting.